### PR TITLE
Fix tests to I/O exercises

### LINKED ISSUE #109: I/O: fix test to `solution_read_write_file`
This test doesn't seem right:

https://github.com/empa-scientific-it/python-tutorial/blob/da295f8f5d4068299143027b566050cfbcefe98b/tutorial/tests/test_input_output.py#L98-L105

### PR DESCRIPTION
**Pull request by yakutovicha**
fixes #109

### CODE REVIEW

File: tutorial/tests/test_input_output.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ def get_data(name: str, data_dir:str="data") -> pl.Path:
     return (pl.Path(current_module.__file__).parent / f"{data_dir}/{name}").resolve()
 
 def reference_solution_find_all_files(f: pl.Path) -> "list[pl.Path]":
-    return list(f.parent.glob(f.name))
+    return list(f.parent.iterdir())
```
Comment (edoardob90) [inline tutorial/tests/test_input_output.py:17]: Also `f.parent.glob("*")` works

Comment (yakutovicha) [reply tutorial/tests/test_input_output.py:17]: yes, they do the same. So any solution works.

**Comment by edoardob90** [comment]
Also `test_print_salutation` doesn't work. `contextlib` does not have a `redirect_stdin` method. I'm not sure if we want to test that exercise as well.

**Review by edoardob90 — CHANGES_REQUESTED** [review]
Also `test_print_salutation` doesn't work. `contextlib` does not have a `redirect_stdin` method. I'm not sure if we want to test that exercise as well.

**Comment by yakutovicha** [comment]
> > > Also `test_print_salutation` doesn't work. `contextlib` does not have a `redirect_stdin` method. I'm not sure how we want to test that exercise.
> > 
> > 
> > This one is not supposed to be auto-tested. @baffelli wrote it in the description.
> 
> Then let's remove the test, because it's not working and not needed

I let @baffelli comment on this.

**Comment by baffelli** [comment]
> > > > Also `test_print_salutation` doesn't work. `contextlib` does not have a `redirect_stdin` method. I'm not sure how we want to test that exercise.
> > > 
> > > 
> > > This one is not supposed to be auto-tested. @baffelli wrote it in the description.
> > 
> > 
> > Then let's remove the test, because it's not working and not needed
> 
> I let @baffelli comment on this.

Indeed we can remove it. `redirect_stdin` was a silly suggestion from copilot. Anyway I could not find an elegant way to capture stdin, therefore we can safely remove the test.

File: tutorial/tests/test_input_output.py (hunk 1 above)
Comment (baffelli) [reply tutorial/tests/test_input_output.py:19]: I did, therefore this is fine. We don't want the solution to be identical :)

**Pull request merged**